torchrl/data/tensordict/tensordict.py

```diff
@@ -2730,7 +2733,8 @@ def set(
         _run_checks: bool = True,
     ) -> TensorDictBase:
         if self.is_locked:
-            raise RuntimeError("Cannot modify immutable TensorDict")
+            if not inplace or key not in self.keys():
```

Comment:
iterating through the keys can be a tedious operation when repeated frequently. We try to avoid doing that twice per function call by creating a set at the beginning. Can you move that before your call and check the presence in the set (rather than in the iterator method)?

Reply:
I moved the local `keys` variable from below up to remove the duplicate call.  Also wrapped other calls of `self.keys()` with `set()` just in case.